wayland-server/src/socket.rs

```diff
@@ -83,18 +82,16 @@ impl ListeningSocket {
         // https://gitlab.freedesktop.org/libbsd/libbsd/-/blob/73b25a8f871b3a20f6ff76679358540f95d7dbfd/src/flopen.c#L71
         loop {
             // open the lockfile
-            let lock_fd = open(
-                &lock_path,
-                OFlag::O_CREAT | OFlag::O_CLOEXEC | OFlag::O_RDWR,
-                Mode::S_IRUSR | Mode::S_IWUSR | Mode::S_IRGRP | Mode::S_IWGRP,
-            )
-            .map_err(|_| BindError::PermissionDenied)?;
-
-            // SAFETY: We have just opened the file descriptor.
-            _lock = unsafe { File::from_raw_fd(lock_fd) };
+            _lock = File::options()
+                .create(true)
+                .read(true)
+                .write(true)
+                .mode(0o660)
+                .open(&lock_path)
+                .map_err(|_| BindError::PermissionDenied)?;
```

Comment:
`CLOEXEC` set by default by the `File` constructor I assume?

Reply:
Yes, it should set `CLOEXEC` regardless of what options are used.